Bottom: df78a11fcb050d8c6c85e5688e1fd4109f8ae8d7
Top:    ed28b1d26d8189be843305c1aa0082ca8248d735
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-02-13 17:40:35 +0000

polygon1.c: Refactor poly_Copy0 to use poly_Create(), rather than open-coding

This ensures all polygon initialisation happens in a common place


---

diff --git a/src/polygon1.c b/src/polygon1.c
index f970086..5f2686a 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -2994,12 +2994,10 @@ poly_Copy0 (POLYAREA ** dst, const POLYAREA * src)
   if (src == NULL)
     return TRUE;
 
-  *dst = (POLYAREA *)calloc (1, sizeof (POLYAREA));
-  if (*dst == NULL)
+  if ((*dst = poly_Create ()) == NULL || !poly_Copy1 (*dst, src))
     return FALSE;
-  (*dst)->contour_tree = r_create_tree (NULL, 0, 0);
 
-  return poly_Copy1 (*dst, src);
+  return TRUE;
 }
 
 void
